openpype/modules/kitsu/utils/update_op_with_zou.py

```diff
@@ -0,0 +1,372 @@
+"""Functions to update OpenPype data using Kitsu DB (a.k.a Zou)."""
+from copy import deepcopy
+from typing import Dict, List
+
+from pymongo import DeleteOne, UpdateOne
+from pymongo.collection import Collection
+import gazu
+from gazu.task import (
+    all_tasks_for_asset,
+    all_tasks_for_shot,
+)
+
+from openpype.pipeline import AvalonMongoDB
+from openpype.api import get_project_settings
+from openpype.lib import create_project
+from openpype.modules.kitsu.utils.credentials import validate_credentials
+
+
+def create_op_asset(gazu_entity: dict) -> dict:
+    """Create OP asset dict from gazu entity.
+
+    :param gazu_entity:
+    """
+    return {
+        "name": gazu_entity["name"],
+        "type": "asset",
+        "schema": "openpype:asset-3.0",
+        "data": {"zou": gazu_entity, "tasks": {}},
+    }
+
+
+def set_op_project(dbcon: AvalonMongoDB, project_id: str):
+    """Set project context.
+
+    Args:
+        dbcon (AvalonMongoDB): Connection to DB.
+        project_id (str): Project zou ID
+    """
+    project = gazu.project.get_project(project_id)
+    project_name = project["name"]
+    dbcon.Session["AVALON_PROJECT"] = project_name
+
+
+def update_op_assets(
+    project_col: Collection,
+    entities_list: List[dict],
+    asset_doc_ids: Dict[str, dict],
+) -> List[Dict[str, dict]]:
+    """Update OpenPype assets.
+    Set 'data' and 'parent' fields.
+
+    Args:
+        project_col (Collection): Mongo project collection to sync
+        entities_list (List[dict]): List of zou entities to update
+        asset_doc_ids (Dict[str, dict]): Dicts of [{zou_id: asset_doc}, ...]
+
+    Returns:
+        List[Dict[str, dict]]: List of (doc_id, update_dict) tuples
+    """
+    project_name = project_col.name
+
+    assets_with_update = []
+    for item in entities_list:
+        # Update asset
+        item_doc = asset_doc_ids[item["id"]]
+        item_data = deepcopy(item_doc["data"])
+        item_data.update(item.get("data") or {})
```

Comment:
Looks like this line will add to into asset document's data more then should. For example `frame_in` and `frame_out`...

Reply:
poped these useless data